src/CommunityToolkit.Maui/ImageSources/GravatarImageSource.shared.cs

```diff
@@ -0,0 +1,252 @@
+﻿namespace CommunityToolkit.Maui.ImageSources;
+
+using System.Diagnostics;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Maui.Controls;
+
+/// <summary>Gravatar image source.</summary>
+/// <remarks>Note that <see cref="UriImageSource"/> is sealed and can't be used as a parent!</remarks>
+public class GravatarImageSource : StreamImageSource
+{
+	/// <summary>The backing store for the <see cref="CacheValidity" /> bindable property.</summary>
+	public static readonly BindableProperty CacheValidityProperty = BindableProperty.Create(nameof(CacheValidity), typeof(TimeSpan), typeof(GravatarImageSource), TimeSpan.FromDays(1));
+
+	/// <summary>The backing store for the <see cref="Size" /> bindable property.</summary>
+	public static readonly BindableProperty SizeProperty = BindableProperty.Create(nameof(Size), typeof(int), typeof(GravatarImageSource), 0, propertyChanged: OnSizePropertyChanged);
+
+	/// <summary>The backing store for the <see cref="CachingEnabled" /> bindable property.</summary>
+	public static readonly BindableProperty CachingEnabledProperty = BindableProperty.Create(nameof(CachingEnabled), typeof(bool), typeof(GravatarImageSource), true);
+
+	/// <summary>The backing store for the <see cref="Email" /> bindable property.</summary>
+	public static readonly BindableProperty? EmailProperty = BindableProperty.Create(nameof(Email), typeof(string), typeof(GravatarImageSource), defaultValue: null, propertyChanged: OnEmailPropertyChanged);
```

Comment:
This doesn't need to be nullable

```suggestion
	public static readonly BindableProperty EmailProperty = BindableProperty.Create(nameof(Email), typeof(string), typeof(GravatarImageSource), defaultValue: null, propertyChanged: OnEmailPropertyChanged);
```

Reply:
Removed.